### PR TITLE
[SYCL][libdevice] Add type cast functions between half and float/integral type

### PR DESCRIPTION
**Pull request by jinge90**

### CODE REVIEW

**Review by xtian-github — APPROVED** [review]
LGTM

**Review by dm-vodopyanov — APPROVED** [review]

**Review by akolesov-intel — APPROVED** [review]

**Comment by jinge90** [comment]
Hi, @pvchupin 
Could you take a look at this PR?

Thanks very much.

**Comment by jinge90** [comment]
Hi, @pvchupin 
Kind ping.

**Comment by pvchupin** [comment]
@jinge90, please get @intel/dpcpp-tools-reviewers approval.

File: libdevice/imf_utils/double_convert.cpp [hunk 1]
```diff
@@ -441,4 +441,13 @@ DEVICE_EXTERN_C_INLINE
 double __devicelib_imf_ull2double_rz(unsigned long long int x) {
   return __ull2double_rz(x);
 }
+
```
Comment (asudarsa) [inline libdevice/imf_utils/double_convert.cpp:444]: I was just curious why only the _rte version is provided for this conversion and _rtp, _rtn and _rtz are not provided.

Thanks

Comment (jinge90) [reply libdevice/imf_utils/double_convert.cpp:444]: Hi, @asudarsa 
This PR aims to provide some util functions to do type cast between half and other types, the motivation is CUDA math provides such functionalities while SYCL doesn't, providing those utils can help users who need to port CUDA sources to SYCL. In CUDA math APIs, NV only provides rte for double to half, this is whey we only provide rte here.

Thanks very much.

**Comment by jinge90** [comment]
Hi, @asudarsa and @pvchupin 
Kind ping~

**Review by asudarsa — APPROVED** [review]
Other than my earlier comment which has been resolved, the change looks good to me. Thanks

**Comment by jinge90** [comment]
Hi, @pvchupin 
Could you take a look if we can merge this patch?

Thanks very much.

**Pull request merged**